File: solutions/python3/0111.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def minDepth(self, root: TreeNode) -> int:
        if not root:
            return 0

        ans = 0
        q = collections.deque([root])

        while q:
            ans += 1
            for i in range(len(q)):
                node = q.popleft()
                if not node.left and not node.right:
                    return ans
                if node.left:
                    q.append(node.left)
                if node.right:
                    q.append(node.right)

        return -1

File: solutions/cpp/0111.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */

class Solution {
public:
    int minDepth(TreeNode* root) {
        if (!root) return 0;

        int ans = 0;
        queue<TreeNode*> queue;
        queue.push(root);

        while (!queue.empty()) {
            ans++;
            int queueSize = queue.size();
            for (int i = 0; i < queueSize; i++) {
                TreeNode* node = queue.front();
                queue.pop();
                if (!node->left && !node->right) return ans;
                if (node->left) queue.push(node->left);
                if (node->right) queue.push(node->right);
            }
        }

        return -1;
    }
};